File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        current = 0
        left = 0
        right = 1
        while( left < right and left < len(prices) and right < len(prices)):
            current = max(current, prices[right]-prices[left])
            if(prices[right] < prices[left]):
                left = right
                right = left + 1
            else:
                right += 1
        return current

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
private: 

    int solve(TreeNode* root, int & ans) {
        if(root == nullptr) return 0;
        int lh = 1 + solve(root->left, ans);
        int rh = 1 + solve(root->right, ans);
        ans = max(ans, lh + rh -1);
        return max(lh, rh);
    }

public:
    int diameterOfBinaryTree(TreeNode* root) {
        int ans = 0;
        int h = solve(root, ans);
        return ans - 1;
    }
};
